2-0-data-structures-and-algorithms/src/test/java/com/bobocode/linked_queue/LinkedQueueTest.java

```diff
@@ -1,103 +1,372 @@
+        Integer secondElement = (Integer) pollElementFromQueue();
+
+        assertThat(size).isEqualTo(2);
+        assertThat(isEmpty).isEqualTo(false);
+        assertThat(firstElement).isEqualTo(12);
+        assertThat(secondElement).isEqualTo(111);
     }
 
     @Test
     @Order(6)
-    void pollElement() {
-        integerQueue.add(33);
-        integerQueue.add(123);
-        integerQueue.add(222);
-        integerQueue.add(444);
-
-        integerQueue.poll(); // should poll 33
+    void addIncreasesQueueSize() {
+        integerQueue.add(1);
+        integerQueue.add(2);
+        int size = this.integerQueue.size();
 
-        assertEquals(123, integerQueue.poll().intValue());
-        assertEquals(2, integerQueue.size());
+        assertThat(size).isEqualTo(2);
     }
 
     @Test
     @Order(7)
-    void pollLastElement() {
-        integerQueue.add(8);
-        integerQueue.add(123);
-        integerQueue.add(99);
-        integerQueue.add(46);
-
-        integerQueue.poll(); // should poll 8
-        integerQueue.poll(); // should poll 123
-        integerQueue.poll(); // should poll 99
+    void isEmptyReturnsFalseWhenElementIsAdded() {
+        integerQueue.add(1);
+        boolean isEmpty = integerQueue.isEmpty();
 
-        assertEquals(46, integerQueue.poll().intValue());
-        assertEquals(0, integerQueue.size());
+        assertThat(isEmpty).isEqualTo(false);
     }
 
     @Test
     @Order(8)
-    void isEmptyOnEmptyQueue() {
-        assertTrue(integerQueue.isEmpty());
+    void pollReturnsNullWhenQueueIsEmpty() {
+        Integer firstElement = this.integerQueue.poll();
+
+        assertThat(firstElement).isEqualTo(null);
     }
 
     @Test
     @Order(9)
-    void isEmpty() {
-        integerQueue.add(3);
-        integerQueue.add(9);
+    void poll() {
+        addIntElementToQueue(11);
+        addIntElementToQueue(111);
+        Integer firstElement = this.integerQueue.poll();
+        Integer secondElement = this.integerQueue.poll();
+        int size = getInternalSize();
+        boolean isEmpty = isEmptyQueue();
+
+        assertThat(size).isEqualTo(0);
+        assertThat(isEmpty).isEqualTo(true);
+        assertThat(firstElement).isEqualTo(11);
+        assertThat(secondElement).isEqualTo(111);
+    }
+
+    @Test
+    @Order(10)
+    void pollDecreasesQueueSize() {
+        addIntElementToQueue(11);
+        addIntElementToQueue(111);
+        this.integerQueue.poll();
+        int size = this.integerQueue.size();
+
+        assertThat(size).isEqualTo(1);
+    }
+
+    @Test
+    @Order(11)
+    void pollMakesSizeZeroWhenQueueHasSingleElement() {
+        addIntElementToQueue(12);
+        Integer element = this.integerQueue.poll();
+        int size = this.integerQueue.size();
+
+        assertThat(size).isEqualTo(0);
+        assertThat(element).isEqualTo(12);
+    }
+
+    @Test
+    @Order(12)
+    void isEmptyReturnsTrueWhenLastElementIsPolled() {
```

Comment:
We should avoid using of several implemented methods. Let's replace `isEmpty()` by reflection and name the method `pollMakesQueueEmptyWhenQueueHasSingleElement()`